src/coreclr/jit/codegenloongarch64.cpp

```diff
@@ -5067,7 +5067,8 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
             assert((tgtBlock->bbTgtStkDepth * sizeof(int) == genStackLevel) || isFramePointerUsed());
 #endif // !FEATURE_FIXED_OUT_ARGS
 
-            emit->emitIns_J(INS_bcnez, tgtBlock, (int)1 /* cc */);
+            instruction ins = treeNode->gtFlags & GTF_JCC_FALSE ? INS_bceqz : INS_bcnez;
+            emit->emitIns_J(ins, tgtBlock, (int)1 /* cc */);
```

Comment:
Something like:
```cpp
GenCondition cc = treeNode->AsCC()->gtCondition;
assert((cc.GetCode() == GenCondition::EQ) || (cc.GetCode() == GenCondition::NE));
instruction ins = (cc.GetCode() == GenCondition::EQ) ? INS_bceqz : INS_bcnez;
```

Reply:
arm64 effectively works in the same way for `GT_JCMP`:
https://github.com/dotnet/runtime/blob/6bb544932858aedf13ed24a01723f57b1fd48b5d/src/coreclr/jit/codegenarm64.cpp#L4879-L4881

Reply:
Yes, this is OK.
but also liking a workround.